core/store/src/trie/mod.rs

```diff
@@ -290,25 +290,25 @@ fn decode_children(cursor: &mut Cursor<&[u8]>) -> Result<[Option<CryptoHash>; 16
 
 impl RawTrieNode {
     fn encode_into(&self, out: &mut Vec<u8>) {
```

Comment:
I actually looked at it as well and had one more thought.  How about adding `suffix: &[u8]` argument to this function so that we can `out.reserve` the correct amount here? https://github.com/near/nearcore/pull/7253

Reply:
Interesting idea.  Left some feedback on the PR itself.

Reply:
Looks like the reserve isn’t beneficial in practice as per your comment there.